systemtest/src/test/java/io/strimzi/systemtest/rollingupdate/AlternativeReconcileTriggersST.java

```diff
@@ -453,6 +458,178 @@ void testAddingAndRemovingJbodVolumes() {
+            }, testStorage.getNamespaceName());
+        }
+
+        // Wait util it rolls
+        brokerPods = RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getBrokerSelector(), 3, brokerPods);
+
+        // verify that Kraft metadata log will be re-assigned to another volume (the minimum id, which is 0 now that's why data-0)
+        verifyKafkaKraftMetadataLog(testStorage, 0, 2);
+
+        resourceManager.createResourceWithWait(clients.consumerStrimzi());
+        ClientUtils.waitForInstantConsumerClientSuccess(testStorage);
+
+        // ##############################
+        // Validate that continuous clients finished successfully
+        // ##############################
+        ClientUtils.waitForContinuousClientSuccess(testStorage, continuousClientsMessageCount);
+        // ##############################
+    }
+
+    /**
+     * Verifies the presence or absence of KRaft metadata logs across specified volume directories within Kafka pods.
+     * This method iterates through all Kafka pods retrieved based on the broker selector from the test storage configuration,
+     * checking each configured volume for the presence of KRaft metadata files. The method asserts that the metadata log
+     * exists only in the specified KRaft metadata volume and not in others, ensuring correct metadata log placement
+     * according to the test configuration.
+     *
+     * @param testStorage               An instance of TestStorage containing configuration and context for the current test,
+     *                                  including namespace and broker selector for identifying relevant Kafka pods.
+     * @param kraftMetadataVolumeId     The volume ID expected to contain the KRaft metadata log. This method will
+     *                                  assert the presence of metadata logs in this volume and their absence in others.
+     * @param numberOfVolumes           The total number of volumes configured in the JBOD (Just a Bunch Of Disks) for each Kafka broker.
+     *                                  This dictates how many volume directories the method will check within each Kafka pod.
+     */
+    private void verifyKafkaKraftMetadataLog(final TestStorage testStorage, final int kraftMetadataVolumeId, final int numberOfVolumes) {
+        final List<Pod> kafkaPods = kubeClient().listPods(testStorage.getNamespaceName(), testStorage.getBrokerSelector());
+        int kafkaIndex = 0; // Ensure this index is managed appropriately if used outside this method context.
+
+        for (final Pod kafkaPod : kafkaPods) {
+            List<String> directories = new ArrayList<>();
+
+            // Dynamically create a list of volume directories to check, based on the total number of volumes.
+            for (int volumeId = 0; volumeId < numberOfVolumes; volumeId++) {
+                directories.add("/var/lib/kafka/data-" + volumeId + "/kafka-log" + kafkaIndex + "/__cluster_metadata-0");
```

Comment:
You can maybe create some helper method to just return the dir path based on some arguments? Because it's almost the same on two places and at least it would not be hardcoded that way

Reply:
Sure :) will do thanks for spotting this